src/core/queryClient.ts

```diff
@@ -102,70 +102,74 @@ export class QueryClient {
     return this.mutationCache.findAll({ ...filters, fetching: true }).length
   }
 
-  getQueryData<TData = unknown>(
+  getQueryData<TQueryFnData = unknown>(
     queryKey: QueryKey,
     filters?: QueryFilters
-  ): TData | undefined {
-    return this.queryCache.find<TData>(queryKey, filters)?.state.data
+  ): TQueryFnData | undefined {
+    return this.queryCache.find<TQueryFnData>(queryKey, filters)?.state.data
   }
 
-  getQueriesData<TData = unknown>(queryKey: QueryKey): [QueryKey, TData][]
-  getQueriesData<TData = unknown>(filters: QueryFilters): [QueryKey, TData][]
-  getQueriesData<TData = unknown>(
+  getQueriesData<TQueryFnData = unknown>(
+    queryKey: QueryKey
+  ): [QueryKey, TQueryFnData][]
+  getQueriesData<TQueryFnData = unknown>(
+    filters: QueryFilters
+  ): [QueryKey, TQueryFnData][]
+  getQueriesData<TQueryFnData = unknown>(
     queryKeyOrFilters: QueryKey | QueryFilters
-  ): [QueryKey, TData][] {
+  ): [QueryKey, TQueryFnData][] {
     return this.getQueryCache()
       .findAll(queryKeyOrFilters)
       .map(({ queryKey, state }) => {
-        const data = state.data as TData
+        const data = state.data as TQueryFnData
```

Comment:
looking at these types and the type assertion, I actually think the types are wrong here. If you have a query that is in `error` state, or that hasn't run yet (e.g. because it is `disabled`), and you find it with `getQueriesData`, you will get `undefined` back. I think we need to at least union the return type with undefined:

```diff
- [QueryKey, TQueryFnData][]
+ [QueryKey, TQueryFnData | undefined][]
```

This would be in line to what `getQueryData`. What do you think? Should we add this here as a fix?